# EDA ANALYSIS - VICTOR HORCAS 


#### Análisis exploratorio de datos sobre un dataset de los mejores atletas de crossfit a nivel mundial. 


### Importamos todas las librerías que vamos a utilizar en este proyecto.

In [2]:
import seaborn as sns
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np

## 1. Exploración de los datos

#### Cargamos el dataset escogido para el análisis

In [ ]:
import kagglehub

In [ ]:
crossfit = kagglehub.dataset_download("tgomesjuliana/crossfit-competitions")
crossfit

In [5]:
crossfit_df = pd.read_csv("D:\\firstprojecteda\\data\\consolidated_athletes_information.csv")

#### Previsualización de forma general de los datos

In [37]:
# Base de datos entera
crossfit_df

competitorId       competitorName firstName     lastName gender  \
0           469656.0        Jeffrey Adler   Jeffrey        Adler      M   
1            34796.0        Scott Panchik     Scott      Panchik      M   
2           105875.0          Travis Mead    Travis         Mead      M   
3           310970.0        Saxon Panchik     Saxon      Panchik      M   
4            11435.0  Richard Froning Jr.   Richard  Froning Jr.      M   
...              ...                  ...       ...          ...    ...   
399337      872275.0     Caroline Stanley  Caroline      Stanley      F   
399338      174547.0      Amanda Barnhart    Amanda     Barnhart      F   
399339      121033.0         Sydney Wells    Sydney        Wells      F   
399340      503582.0        Alexis Raptis    Alexis       Raptis      F   
399341     1547449.0            Emma Cary      Emma         Cary      F   

       countryOfOriginName     regionName  affiliateId  \
0                   Canada  North America      18059.0   
1            United States  North America       7991.0   
2            United States  North America       9155.0   
3            United States  North America      22505.0   
4            United States  North America       3220.0   
...                    ...            ...          ...   
399337       United States  North America       3617.0   
399338       United States  North America      14583.0   
399339       United States  North America      19593.0   
399340       United States  North America      25335.0   
399341       United States  North America       1893.0   

                  affiliateName   age height  weight  overallRank  \
0           CrossFit Wonderland  27.0  69 in  197 lb          1.0   
1            CrossFit Mentality  33.0  69 in  187 lb          2.0   
2          Iron Valley CrossFit  34.0  73 in  205 lb          3.0   
3            CrossFit Cliffside  25.0  69 in  180 lb          4.0   
4               CrossFit Mayhem  33.0  69 in  194 lb          5.0   
...                         ...   ...    ...     ...          ...   
399337    College Hill CrossFit  23.0  63 in  145 lb          NaN   
399338       CrossFit High Gear  31.0  67 in  160 lb          NaN   
399339  CrossFit East Nashville  28.0  66 in  151 lb          NaN   
399340             TTT CrossFit  24.0  65 in  155 lb          NaN   
399341          CrossFit Naples  19.0  64 in  140 lb          NaN   

        overallScore  year competition  height_cm  weight_kg  
0              101.0  2021        open      176.0       90.0  
1              141.0  2021        open      176.0       85.0  
2              165.0  2021        open      186.0       93.0  
3              217.0  2021        open      176.0       82.0  
4              254.0  2021        open      176.0       88.0  
...              ...   ...         ...        ...        ...  
399337           0.0  2023       games      161.0       66.0  
399338           0.0  2023       games      171.0       73.0  
399339           0.0  2023       games      168.0       69.0  
399340           0.0  2023       games      166.0       71.0  
399341           0.0  2023       games      163.0       64.0  

[399342 rows x 18 columns]

In [31]:
# 5 primeras filas
crossfit_df.head (5)

competitorId       competitorName firstName     lastName gender  \
0      469656.0        Jeffrey Adler   Jeffrey        Adler      M   
1       34796.0        Scott Panchik     Scott      Panchik      M   
2      105875.0          Travis Mead    Travis         Mead      M   
3      310970.0        Saxon Panchik     Saxon      Panchik      M   
4       11435.0  Richard Froning Jr.   Richard  Froning Jr.      M   

  countryOfOriginName     regionName  affiliateId         affiliateName   age  \
0              Canada  North America      18059.0   CrossFit Wonderland  27.0   
1       United States  North America       7991.0    CrossFit Mentality  33.0   
2       United States  North America       9155.0  Iron Valley CrossFit  34.0   
3       United States  North America      22505.0    CrossFit Cliffside  25.0   
4       United States  North America       3220.0       CrossFit Mayhem  33.0   

  height  weight  overallRank  overallScore  year competition  height_cm  \
0  69 in  197 lb          1.0         101.0  2021        open      176.0   
1  69 in  187 lb          2.0         141.0  2021        open      176.0   
2  73 in  205 lb          3.0         165.0  2021        open      186.0   
3  69 in  180 lb          4.0         217.0  2021        open      176.0   
4  69 in  194 lb          5.0         254.0  2021        open      176.0   

   weight_kg  
0       90.0  
1       85.0  
2       93.0  
3       82.0  
4       88.0

Visualizamos información básica de nuestro dataset y los tipos de datos que contiene.

In [7]:
crossfit_df.describe()

competitorId    affiliateId            age    overallRank  \
count  3.993420e+05  343166.000000  399342.000000  399262.000000   
mean   1.249626e+06   13407.674816      35.036841   55504.236694   
std    7.227477e+05    8562.634020       8.101685   42532.260165   
min    8.600000e+01       4.000000      16.000000      -1.000000   
25%    6.559250e+05    5550.000000      29.000000   17732.000000   
50%    1.239003e+06   13295.000000      34.000000   48394.000000   
75%    1.942362e+06   20161.000000      41.000000   87406.750000   
max    2.523369e+06   32333.000000      54.000000  157742.000000   

        overallScore           year      height_cm      weight_kg  
count  399342.000000  399342.000000  399342.000000  399342.000000  
mean   203432.557287    2022.035338     173.984727      77.803296  
std    143953.597002       0.820031       9.333170      13.075751  
min         0.000000    2021.000000     150.000000      50.000000  
25%     79336.000000    2021.000000     168.000000      68.000000  
50%    186778.000000    2022.000000     175.000000      79.000000  
75%    307736.000000    2023.000000     181.000000      87.000000  
max    582777.000000    2023.000000     200.000000     110.000000

Haciendo uso de la función "describe" podemos ver algunos datos relevantes para nuestro análisis como puede ser la media de edad (unos 35 años), el período de años mínimo y máximo en el que está basado este dataset (2021 - 2023), la altura (174 cm) y peso medios (78 kg) y algunos datos más específicos como son los percentiles 25, 50 y 75 que se utilizan para indicar la posición de un valor en relación con el resto de los datos.

In [32]:
crossfit_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 399342 entries, 0 to 399341
Data columns (total 18 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   competitorId         399342 non-null  float64
 1   competitorName       399340 non-null  object 
 2   firstName            399341 non-null  object 
 3   lastName             399338 non-null  object 
 4   gender               399342 non-null  object 
 5   countryOfOriginName  396587 non-null  object 
 6   regionName           399342 non-null  object 
 7   affiliateId          343166 non-null  float64
 8   affiliateName        343166 non-null  object 
 9   age                  399342 non-null  float64
 10  height               399342 non-null  object 
 11  weight               399342 non-null  object 
 12  overallRank          399262 non-null  float64
 13  overallScore         399342 non-null  float64
 14  year                 399342 non-null  int64  
 15  competition      

In [33]:
crossfit_df.dtypes

competitorId           float64
competitorName          object
firstName               object
lastName                object
gender                  object
countryOfOriginName     object
regionName              object
affiliateId            float64
affiliateName           object
age                    float64
height                  object
weight                  object
overallRank            float64
overallScore           float64
year                     int64
competition             object
height_cm              float64
weight_kg              float64
dtype: object

Con estas dos funciones obtenemos un recuento de valores no nulos y vemos el tipo de datos que tenemos (float, object e integer).

In [6]:
crossfit_df.columns

Index(['competitorId', 'competitorName', 'firstName', 'lastName', 'gender',
       'countryOfOriginName', 'regionName', 'affiliateId', 'affiliateName',
       'age', 'height', 'weight', 'overallRank', 'overallScore', 'year',
       'competition', 'height_cm', 'weight_kg'],
      dtype='object')

Aquí tenemos de un vistazo rápido todas las columnas (variables) que posee nuestro dataset.

Podemos ver que existen dos variables para la altura y dos para el peso (se considera eliminar una de cada una)

#### VALORES NULOS 

In [34]:
# Sumamos los valores nulos por cada columna
crossfit_df.isna().sum() 

competitorId               0
competitorName             2
firstName                  1
lastName                   4
gender                     0
countryOfOriginName     2755
regionName                 0
affiliateId            56176
affiliateName          56176
age                        0
height                     0
weight                     0
overallRank               80
overallScore               0
year                       0
competition                0
height_cm                  0
weight_kg                  0
dtype: int64

In [ ]:
# Calculo el porcentaje de valores nulos por columna para considerar si elimino la fila completa o trato esos valores nulos
porcentaje_nulos = crossfit_df.isna().mean() * 100
print(porcentaje_nulos)

competitorId            0.000000
competitorName          0.000501
firstName               0.000250
lastName                0.001002
gender                  0.000000
countryOfOriginName     0.689885
regionName              0.000000
affiliateId            14.067140
affiliateName          14.067140
age                     0.000000
height                  0.000000
weight                  0.000000
overallRank             0.020033
overallScore            0.000000
year                    0.000000
competition             0.000000
height_cm               0.000000
weight_kg               0.000000
dtype: float64


El dataset escogido no cuenta con gran cantidad de datos nulos, en especial, en las columnas más necesarias para nuestro proyecto como son el "competitorId", "gender", "age" y "overallScore". Las columnas con más valores nulos se podría considerar eliminarlas, ya que, aportan identificadores poco relevantes para nuestro proyecto.

#### VALORES DUPLICADOS 

In [9]:
crossfit_df.duplicated().sum()

np.int64(0)

Este dataset no contiene valores duplicados

#### VALORES ÚNICOS 

In [12]:
print(crossfit_df["gender"].unique())

['M' 'F']


Para la variable género nos interesa que sólo existan dos valores únicos, lo cuál si que ocurre en esta ocasión.

M --> Male

F --> Female

In [13]:
print(crossfit_df["year"].unique())

[2021 2022 2023]


Para la variable año sólo existen tres valores únicos, ya que, esta base de datos recopila valores entre el año 2021 - 2023.

#### VALORES ATÍPICOS 

In [14]:
print(crossfit_df[crossfit_df["overallScore"] < 0])

Empty DataFrame
Columns: [competitorId, competitorName, firstName, lastName, gender, countryOfOriginName, regionName, affiliateId, affiliateName, age, height, weight, overallRank, overallScore, year, competition, height_cm, weight_kg]
Index: []


La variable "overallScore" no presenta valores atípicos.

In [15]:
print(crossfit_df[crossfit_df["overallRank"] < 0])

        competitorId    competitorName firstName   lastName gender  \
257361      458446.0  Nasser Alruwayeh    Nasser  Alruwayeh      M   
257412      879211.0         Phil Toon      Phil       Toon      M   
257474      489030.0     Nycolas Joyal   Nycolas      Joyal      M   
257584      799012.0     Pedro Martins     Pedro    Martins      M   
257614      261255.0  Claudia Espinosa   Claudia   Espinosa      F   

       countryOfOriginName     regionName  affiliateId     affiliateName  \
257361              Kuwait           Asia       4369.0       CrossFit Q8   
257412       United States  North America       1893.0   CrossFit Naples   
257474              Canada  North America          NaN               NaN   
257584              Brazil  South America      15173.0  CrossFit Barueri   
257614             Ecuador  South America          NaN               NaN   

         age  height  weight  overallRank  overallScore  year competition  \
257361  29.0  183 cm  105 kg         -1.0    

In [ ]:
atypical_overall_rank = crossfit_df[crossfit_df["overallRank"] < 0][["competitorId", "competitorName"]]
atypical_overall_rank

competitorId    competitorName
257361      458446.0  Nasser Alruwayeh
257412      879211.0         Phil Toon
257474      489030.0     Nycolas Joyal
257584      799012.0     Pedro Martins
257614      261255.0  Claudia Espinosa

La variable "overallRank" sí presenta valores atípicos. He filtrado y creado una nueva variable para los atletas que tienen una posición negativa en el ranking para saber cuáles son.

Como solo son 5 atletas se podría considerar en un futuro eliminarlos para poder realizar un buen análisis en torno a esta variable.

## 2. Limpieza y transformación de los datos

In [26]:
# Se podrían eliminar las columnas de height y weigh, ya que las tenemos duplicadas (nos interesan mas las que están en kg y cm)

# Considerar eliminar las columnas first name y last name y dejar solo el competitor name

# Tratar columna de "overallRank"